src/Microsoft.ML.Data/DataLoadSave/Text/TextLoader.cs

```diff
@@ -1479,16 +1551,38 @@ void ICanSaveModel.Save(ModelSaveContext ctx)
             for (int index = 0; index < memberInfos.Length; index++)
             {
                 var memberInfo = memberInfos[index];
-                var mappingAttr = memberInfo.GetCustomAttribute<LoadColumnAttribute>();
 
-                if (mappingAttr == null)
-                    throw host.Except($"{(memberInfo is FieldInfo ? "Field" : "Property")} '{memberInfo.Name}' is missing the {nameof(LoadColumnAttribute)} attribute");
+                var ignoreLoadColumnAttr = memberInfo.GetCustomAttribute<LoadColumnIgnoreAttribute>();
+                if (ignoreLoadColumnAttr != null && ignoreLoadColumnAttr.Ignore)
+                    continue;
+
+                var mappingLoadColumnAttr = memberInfo.GetCustomAttribute<LoadColumnAttribute>();
+                var mappingLoadColumnNameAttr = memberInfo.GetCustomAttribute<LoadColumnNameAttribute>();
+                var memberInfoType = memberInfo is FieldInfo ? "Field" : "Property";
+
+                if (!hasHeader && mappingLoadColumnAttr == null)
+                    throw host.Except(
+                        $"You should set {nameof(hasHeader)} to true in order to use {nameof(LoadColumnNameAttribute)}. " +
+                        $"To automatically map the name of {memberInfoType} to {nameof(LoadColumnNameAttribute)}, set {nameof(autoMapLoadColumns)} to true. " +
+                        $"To specify a fixed column index, use {nameof(LoadColumnAttribute)}. " +
+                        $"Invalid {memberInfoType} name: {memberInfo.Name}.");
+
+                if (!autoMapLoadColumns && mappingLoadColumnAttr == null && mappingLoadColumnNameAttr == null)
+                    throw host.Except(
+                        $"You shoud set {nameof(autoMapLoadColumns)} to true in order to automatically " +
+                        $"map the name of {memberInfoType} to {nameof(LoadColumnNameAttribute)}. " +
+                        $"To ignore this member, use {nameof(LoadColumnIgnoreAttribute)}. " +
+                        $"Invalid {memberInfoType} name: {memberInfo.Name}.");
 
                 var mappingAttrName = memberInfo.GetCustomAttribute<ColumnNameAttribute>();
 
                 var column = new Column();
                 column.Name = mappingAttrName?.Name ?? memberInfo.Name;
-                column.Source = mappingAttr.Sources.ToArray();
+                column.Source =  (mappingLoadColumnNameAttr != null)
+                    ? mappingLoadColumnNameAttr.Sources.Select(x => new Range(x)).ToArray()
```

Comment:
What happens if two or more of LoadColumn, LoadColumnName and ColumnName are specified and there is a conflict? Can you please identify that and throw an exception.